cmd/thanos/receive.go

```diff
@@ -193,29 +193,28 @@ func runReceive(
 	{
 		// TSDB.
 		cancel := make(chan struct{})
-		// Before actually starting, we need to make sure
-		// the WAL is flushed.
-		startTimeMargin := int64(2 * time.Duration(tsdbCfg.MinBlockDuration).Seconds() * 1000)
-		if err := db.Open(); err != nil {
-			return errors.Wrap(err, "opening storage")
-		}
-		if err := db.Flush(); err != nil {
-			return errors.Wrap(err, "flushing storage")
-		}
 		g.Add(func() error {
 			defer close(dbReady)
 			defer close(uploadC)
 
+			// Before actually starting, we need to make sure the
+			// WAL is flushed. The WAL is flushed after the
+			// hashring ring is loaded.
+			startTimeMargin := int64(2 * time.Duration(tsdbCfg.MinBlockDuration).Seconds() * 1000)
```

Comment:
IMO the run groups should be as much about the action of running something as possible, with these config parameters out of the way. Can we move this variable definition out of the run group but keep it in the block? i.e. line 195?